server/instance_cloud.go

```diff
@@ -165,7 +165,7 @@ func (jci jiraCloudInstance) parseHTTPRequestJWT(r *http.Request) (*jwt.Token, s
 		// HMAC secret is a []byte
 		return []byte(jci.AtlassianSecurityContext.SharedSecret), nil
 	})
-	if err != nil {
+	if err != nil || !token.Valid {
```

Comment:
It will return an error if not valid, but just for sanity. (they check this in their docs)